dask/dataframe/core.py

```diff
@@ -747,6 +747,100 @@ def to_csv(self, filename, **kwargs):
         from .io import to_csv
         return to_csv(self, filename, **kwargs)
 
+    @wraps(pd.DataFrame._get_numeric_data)
+    def _get_numeric_data(self, how='any', subset=None):
+        return map_partitions(pd.DataFrame._get_numeric_data, self.columns, self)
+
+    def _validate_axis(self, axis=0):
+        if axis not in (0, 1, 'index', 'columns', None):
+            raise ValueError('No axis named {0}'.format(axis))
+        # convert to numeric axis
+        return {None: 0, 'index': 0, 'columns': 1}.get(axis, axis)
+
+    @wraps(pd.DataFrame.sum)
+    def sum(self, axis=None):
+        axis = self._validate_axis(axis)
+        if axis == 1:
+            f = lambda x: x.sum(axis=1)
+            return self.map_partitions(f, self.columns, return_type=Series)
```

Comment:
Hrm, the `columns=` kwarg here is troublesome.  The result should be a `Series` with `name=None`.  Currently `map_partitions` is a bit broken in that it only accepts a `columns=` kwarg and defaults to `self.columns` if that value is None.